examples/megacli.cpp

```diff
@@ -114,6 +114,102 @@ int responseprogress = -1;
 //2FA pin attempts
 int attempts = 0;
 
+// sync config used when creating a new sync
+static SyncConfig syncConfig;
+
+// converts the sync config stored in `sync` to a string
+static std::string syncConfigToString(const Sync& sync)
+{
+    auto getOptions = [](const Sync& sync)
+    {
+        std::string desc;
+        desc += ", syncDeletions ";
+        desc += sync.syncDeletions() ? "ON" : "OFF";
+        desc += ", forceOverwrite ";
+        desc += sync.forceOverwrite() ? "ON" : "OFF";
+        return desc;
+    };
+
+    std::string description;
+    if (sync.isUpSync() && sync.isDownSync())
+    {
+        description = "TWOWAY";
+    }
+    else if (sync.isUpSync())
+    {
+        description = "UP";
+        description += getOptions(sync);
+    }
+    else
+    {
+        description = "DOWN";
+        description += getOptions(sync);
+    }
+    return description;
+}
+
+// converts the given sync config to a string
+static std::string syncConfigToString(const SyncConfig& config)
+{
+    auto getOptions = [](const SyncConfig& config)
+    {
+        std::string desc;
+        desc += ", syncDeletions ";
+        desc += config.syncDeletions ? "ON" : "OFF";
+        desc += ", forceOverwrite ";
+        desc += config.forceOverwrite ? "ON" : "OFF";
+        return desc;
+    };
+
+    std::string description;
+    if (config.syncType & SyncConfig::TYPE_UP && config.syncType & SyncConfig::TYPE_DOWN)
+    {
+        description = "TWOWAY";
+    }
+    else if (config.syncType & SyncConfig::TYPE_UP)
+    {
+        description = "UP";
+        description += getOptions(config);
+    }
+    else
+    {
+        description = "DOWN";
+        description += getOptions(config);
+    }
+    return description;
+}
+
+// creates a SyncConfig object from config options as strings.
+// returns a pair where `first` is success and `second` is the sync config.
+static std::pair<bool, SyncConfig> syncConfigFromStrings(std::string type, std::string syncDel = {}, std::string overwrite = {})
+{
+    auto toLower = [](std::string& s)
+    {
+        std::transform(s.begin(), s.end(), s.begin(),
+                       [](const char c){ return std::tolower(c); });
+    };
+
+    toLower(type);
+    toLower(syncDel);
+    toLower(overwrite);
+
+    SyncConfig config;
+    if (type == "up") {
+        config.syncType = SyncConfig::TYPE_UP;
+    } else if (type == "down") {
+        config.syncType = SyncConfig::TYPE_DOWN;
+    } else if (type == "twoway") {
+        config.syncType = SyncConfig::TYPE_DEFAULT;
+    } else {
+        return std::make_pair(false, config);
+    }
+
+    config.syncDeletions = syncDel == "on";
+    config.forceOverwrite = overwrite == "on";
```

Comment:
I tested with `1` instead of `0` without looking at the code, and MEGAcli said:
> Successfully applied new sync config!
Where it is not true. Why not to check the input is correct as you do for the type? It makes me think that perhaps there should be a smarter interface to setup the SyncConfig, whether at the `Sync` object or the `SyncConfig`, in order to ensure its consistency. Maybe I'm too picky, but I think it could be useful later on for your developments.

Reply:
Should be fixed now.